src/wallet/wallet.cpp

```diff
@@ -2467,7 +2477,13 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             }
             setPresetCoins.insert(coin);
         } else {
-            return false; // TODO: Allow non-wallet inputs
+            CTxOut txout;
+            if (coin_control.GetExternalOutput(outpoint, txout)) {
+                nValueFromPresetInputs += txout.nValue;
+                setPresetCoins.insert(CInputCoin(outpoint, txout));
```

Comment:
I think you need to subtract the coin value from `value_to_select`. This looks like a rebasing error.

Reply:
Would be good to add a functional test which funds a transaction using external inputs, asserting that the wallet balance would be insufficient without the external inputs. (As in https://github.com/ElementsProject/elements/pull/755/ which is a rough analogue of this PR.)

Reply:
Should be fixed. Also added a test.